Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/PixelReader.h

```diff
@@ -218,19 +219,25 @@ void FPixelReader::SendPixelsInRenderThread(TSensor &Sensor, bool use16BitFormat
                 }
                 #endif
 
-                // network
-                SCOPE_CYCLE_COUNTER(STAT_CarlaSensorStreamSend);
-                TRACE_CPUPROFILER_EVENT_SCOPE_STR("Stream Send");
-                Stream.Send(Sensor, BufView);
+                if (Sensor.AreClientsListening()) 
+                {
+                  // network
+                  SCOPE_CYCLE_COUNTER(STAT_CarlaSensorStreamSend);
+                  TRACE_CPUPROFILER_EVENT_SCOPE_STR("Stream Send");
+                  Stream.Send(Sensor, BufView);
+                }
               }
             }
           };
 
-          WritePixelsToBuffer(
-              *Sensor.CaptureRenderTarget,
-              carla::sensor::SensorRegistry::get<TSensor *>::type::header_offset,
-              InRHICmdList,
-              std::move(FuncForSending));
+          if (Sensor.AreClientsListening()) 
```

Comment:
**Suggested changes**
Mi recommendation is to move this if to the line 121 of this file.

** Resulting Code**
Line 121 will result like this:
```c++
if (!Sensor.IsPendingKill() && Sensor.AreClientsListening())
```

**Benefit**
- More readable code.
- A bit of performance improvement due we are avoiding more code to be executed when there are no clients listening.
- Will be easier to modify the ode in the future.